### PR TITLE
feat(custom timezoneOffset): Use timezoneOffset from DateAdapter

### LINKED ISSUE #48: Add "timezoneOffset" to DateAdapter instead of getting it from the date object
I'm using a custom DateAdapter to force a specific timezone (in my case UTC without using momentjs), however, due to a few places in this module calling getTimezoneOffset directly from the date object, it breaks on days with a DST change.

Moving it to the DateAdapter will let the library consumer define a timezone.

### PR DESCRIPTION
**Pull request by Kleyguerth**
Closes #48

### CODE REVIEW

File: src/date-adapters/date-adapter/index.ts [hunk 1]
```diff
@@ -62,4 +62,6 @@ export interface DateAdapter {
       weekStartsOn?: number;
     }
   ): Date;
+
+  timezoneOffset(date: Date | number): number;
```
Comment (mattlewis92) [inline src/date-adapters/date-adapter/index.ts]: Let's call it `getTimezoneOffset` to be consistent with other methods

File: src/date-adapters/moment/index.ts [hunk 2]
```diff
@@ -117,6 +117,10 @@ export function adapterFactory(moment): DateAdapter {
     return moment(date).get('minutes');
   }
 
+  function timezoneOffset(date: Date | number): number {
+    return new Date(date).getTimezoneOffset();
```
Comment (mattlewis92) [inline src/date-adapters/moment/index.ts]: I think moment has an api for this, so we can do `moment(date).utcOffset()`

File: src/date-adapters/date-fns/index.ts [hunk 3]
```diff
@@ -26,6 +26,10 @@ import {
 } from 'date-fns';
 import { DateAdapter } from '../date-adapter';
 
+function timezoneOffset(date: Date | number): number {
```
Comment (mattlewis92) [inline src/date-adapters/date-fns/index.ts]: Can we add a test here: https://github.com/mattlewis92/calendar-utils/blob/master/test/calendar-utils-dst.spec.ts

For this method in the date-fns + moment date adapters just to be certain the result is the same for both adapters

**Review by mattlewis92 — CHANGES_REQUESTED** [review]
Looking great, thanks for the quick turnaround! Added a few suggestions 😄

File: src/date-adapters/date-fns/index.ts (hunk 3 above)
Comment (Kleyguerth) [reply src/date-adapters/date-fns/index.ts]: I'm confused about what the test should do. Is it simply passing a date to both adapters and making sure both return the same offset?

Comment (mattlewis92) [reply src/date-adapters/date-fns/index.ts]: > I'm confused about what the test should do. Is it simply passing a date to both adapters and making sure both return the same offset?

Yep, exactly that! By putting it in that file, the timezone offset should be `+60` so you can verify it works for both adapters 😄

**Review by mattlewis92 — APPROVED** [review]
LGTM, great work! Thanks for your contribution!

**Pull request merged**